packages/crayons-core/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Component,
+  Prop,
+  h,
+  Host,
+  Event,
+  EventEmitter,
+  Method,
+  State,
+} from '@stencil/core';
+@Component({
+  tag: 'fw-pagination',
+  styleUrl: 'pagination.scss',
+  shadow: true,
+})
+export class Pagination {
+  @State() end;
+  @State() start;
+
+  /**
+   * The current page number.
+   */
+  @Prop() page = 1;
+  /**
+   * The total number of records.
+   */
+  @Prop({ mutable: true }) totalRecords: number;
+  /**
+   *The number of records to be shown per page. Defaults to 10.
+   */
+  @Prop() recordsPerPage = 10;
+  /**
+   * Triggered when either previous or next button is clicked.
+   */
+  @Event() fwChange: EventEmitter;
+
+  @Method()
+  async previous() {
+    this.goToPrevious();
+  }
+
+  @Method()
+  async next() {
+    this.goToNext();
+  }
+
+  componentWillLoad() {
+    // If both totalRecords and page are passed in but if the start record of current page comes out to be
+    // greater than total number of records then assign start to the last page's first record number.
+    this.start = this.totalRecords
+      ? (this.page - 1) * this.recordsPerPage + 1 > this.totalRecords
+        ? this.totalRecords - this.recordsPerPage + 1
+        : (this.page - 1) * this.recordsPerPage + 1
+      : (this.page - 1) * this.recordsPerPage + 1;
+    this.end = this.start + this.recordsPerPage - 1;
+    this.totalRecords = this.totalRecords || this.end;
+  }
+
+  private goToPrevious() {
+    this.start = Math.max(this.start - this.recordsPerPage, 1);
+    this.end =
+      this.start - this.end !== this.recordsPerPage
+        ? this.start + this.recordsPerPage - 1
+        : this.end - this.recordsPerPage;
+
+    this.fwChange.emit({ startRecord: this.start, endRecord: this.end });
+  }
+
+  private goToNext() {
+    this.start += this.recordsPerPage;
+    this.end = Math.min(this.end + this.recordsPerPage, this.totalRecords);
+
+    this.fwChange.emit({ startRecord: this.start, endRecord: this.end });
+  }
+
+  render() {
+    return (
+      <Host>
+        <div class='current-record'>
+          <span class='record'>{this.start}</span> to{' '}
```

Comment:
@parsuram-vijaysankar , here only the texts `to` and `of` have to be localised.
we can use the same decorator approach here for `to` and `of` text with a default value and a key. 
```
@i18n( defaultValue='of' , key='pagination.of')
@State
ofText = ''
```
and this can be used like `{ this.ofText }`